vllm/utils.py

```diff
@@ -1512,3 +1513,40 @@ def weak_ref_tensors(
     if isinstance(tensors, tuple):
         return tuple(weak_ref_tensor(t) for t in tensors)
     raise ValueError("Invalid type for tensors")
+
+
+def is_in_doc_build() -> bool:
+    try:
+        from sphinx.ext.autodoc.mock import _MockModule
+        return isinstance(torch, _MockModule)
+    except ModuleNotFoundError:
+        return False
+
+
+vllm_lib = Library("vllm", "FRAGMENT")
+
+
+def direct_register_custom_op(
+    library_name: str,
+    op_name: str,
+    op_func: Callable,
+    mutates_args: List[str],
+    fake_impl: Optional[Callable] = None,
+):
```

Comment:
Maybe better to remove the `library_name` argument for now, and maybe rename the function to `direct_register_custom_op_to_vllm_library` for now? 

BTW what is the library_name for?

Reply:
library name is the name for `torch.ops.library_name.op_name`

Reply:
I think keeping it this way and asserting `library_name == "vllm"` is better. In the future, when the bug is solved, we can just change the code inside this function.

Reply:
renamed the argument!